dropwizard-hibernate/src/main/java/io/dropwizard/hibernate/AbstractDAO.java

```diff
@@ -22,26 +20,22 @@
  * @param <E> the class which this DAO manages
  */
 public class AbstractDAO<E> {
-    private final SessionFactory sessionFactory;
     private final Class<?> entityClass;
 
     /**
-     * Creates a new DAO with a given session provider.
-     *
-     * @param sessionFactory    a session provider
+     * Creates a new DAO for {@code <E>}.
      */
-    public AbstractDAO(SessionFactory sessionFactory) {
-        this.sessionFactory = requireNonNull(sessionFactory);
+    public AbstractDAO() {
         this.entityClass = Generics.getTypeParameter(getClass());
     }
 
     /**
-     * Returns the current {@link Session}.
+     * Returns the current {@link Session} according to {@link UnitOfWorkAspect}.
      *
      * @return the current session
      */
     protected Session currentSession() {
-        return sessionFactory.getCurrentSession();
+        return UnitOfWorkAspect.getCurrentSession();
```

Comment:
I believe that this change requires some architectural insight that I do not have; I am not sure if the intent of the dropwizard-hibernate module is to be centered around UnitOfWork. It appears to me that it currently is geared that way and I think that makes a lot of sense - but this change creates a literal coupling that did not previously exist.

There are certainly a number of options that could be explored to offer both a DAO implementation that requires a session factory to be provided and one that is driven off of the context.

But, if the goal is to direct clients toward using UnitOfWork for configurations than perhaps it is best to stick to a single DAO.